src/auth_service/claims_repository/rest/access_router.py

```diff
@@ -57,6 +65,151 @@
+            description="The internal ID of the user",
+        ),
+    ] = None,
+    iva_id: Annotated[
+        str | None,
+        Query(
+            ...,
+            alias="iva_id",
+            description="The ID of the IVA",
+        ),
+    ] = None,
+    dataset_id: Annotated[
+        str | None,
+        Query(
+            ...,
+            alias="dataset_id",
+            description="The ID of the dataset",
+        ),
+    ] = None,
+    valid: Annotated[
+        bool | None,
+        Query(
+            ...,
+            alias="valid",
+            description="Whether the grant is currently valid",
+        ),
+    ] = None,
+    # internal service, authorization without token via service mesh
+) -> list[Grant]:
+    """Get download access grants.
+
+    You can filter the grants by user ID, IVA ID, and dataset ID
+    and by whether the grant is currently valid or not.
+    """
+    # Determine all controlled access grants for the user
+    grants: list[Grant] = []
+    users: dict[str, User | None] = {}  # user cache
+    mapping = create_controlled_access_filter(
+        user_id=user_id, iva_id=iva_id, dataset_id=dataset_id
+    )
+    async for claim in claim_dao.find_all(mapping=mapping):
+        if claim.revocation_date:
+            continue  # revoked claims should be considered deleted as grants
+        if valid is not None and is_valid_claim(claim) != valid:
+            continue  # filter by validity if requested
+        dataset_id = dataset_id_for_download_access(claim)
+        if not dataset_id:
+            continue  # consider only claims for the right source
+        # find user name and email
+        user_id = claim.user_id
+        try:
+            user = users[user_id]
+        except KeyError:
+            try:
+                user = await user_dao.get_by_id(user_id)
+            except ResourceNotFoundError:
+                user = None
+            users[user_id] = user
+        if not user:
+            continue
+        grants.append(
+            Grant(
+                id=claim.id,
+                user_id=user_id,
+                iva_id=claim.iva_id,
+                dataset_id=dataset_id,
+                created=claim.creation_date,
+                valid_from=claim.valid_from,
+                valid_until=claim.valid_until,
+                user_name=user.name,
+                user_email=user.email,
+                user_title=user.title,
+            )
+        )
+    # sort the output by ID to make it reproducible
+    return sorted(grants, key=attrgetter("id"))
+
+
+@start_span()
+@router.delete(
+    "/download-access/grants/{grant_id}",
+    operation_id="delete_download_access_grant",
+    tags=TAGS,
+    summary="Delete a download access grant",
+    description="Endpoint to delete an existing download access grant. This will revoke the corresponding claim.",
+    responses={
+        204: {
+            "description": "Access grant have been deleted.",
```

Comment:
Depending on verbiage:
```suggestion
            "description": "Access grant has been deleted.",
```
or 
```suggestion
            "description": "Access grant has been revoked.",
```

Reply:
Changed.